### PR TITLE
Add missing test path for manifest during WPT syncing.

### LINKED ISSUE #27223: WPT sync is busted
```
Error running mach:
    ['update-wpt', '--sync', '--no-upstream', '--patch']
The error occurred in code that was called by the mach command. This is either
a bug in the called code itself or in the way that mach is calling it.
You can invoke |./mach busted| to check if this issue is already on file. If it
isn't, please use |./mach busted file| to report it. If |./mach busted| is
misbehaving, you can also inspect the dependencies of bug 1543241.
If filing a bug, please include the full output of mach, including this error
message.
The details of the failure are as follows:
IOError: [Errno 2] No such file or directory: u'/element-timing/observe-svg-image.html'
  File "/repo/python/servo/testing_commands.py", line 514, in update_wpt
    return run_globals["update_tests"](**kwargs)
  File "/repo/tests/wpt/update.py", line 23, in update_tests
    rv = update.run_update(logger, **kwargs)
  File "/repo/tests/wpt/update/__init__.py", line 26, in run_update
    return updater.run()
  File "/repo/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/update/update.py", line 184, in run
    rv = update_runner.run()
  File "/repo/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/update/base.py", line 61, in run
    rv = step(self.logger).run(step_index, self.state)
  File "/repo/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/update/base.py", line 32, in run
    self.create(state)
  File "/repo/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/update/update.py", line 77, in create
    runner.run()
  File "/repo/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/update/base.py", line 61, in run
    rv = step(self.logger).run(step_index, self.state)
  File "/repo/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/update/base.py", line 32, in run
    self.create(state)
  File "/repo/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/update/sync.py", line 111, in create
    update.update(state.sync["path"], state.test_manifest)
  File "/repo/tests/wpt/sync/tools/manifest/update.py", line 38, in update
    return manifest.update(tree, parallel)
  File "/repo/tests/wpt/sync/tools/manifest/manifest.py", line 249, in update
    for result in results:
  File "/usr/lib/python2.7/multiprocessing/pool.py", line 289, in <genexpr>
    return (item for chunk in result for item in chunk)
  File "/usr/lib/python2.7/multiprocessing/pool.py", line 673, in next
    raise value
```
I suspect https://github.com/web-platform-tests/wpt/pull/23733.

### PR DESCRIPTION
**Pull request by jdm**
Fixes #27223.

### CODE REVIEW

File: tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/update/sync.py [hunk 1]
```diff
@@ -100,7 +100,7 @@ class LoadManifest(Step):
     def create(self, state):
         from manifest import manifest
         state.manifest_path = os.path.join(state.metadata_path, "MANIFEST.json")
-        state.test_manifest = manifest.Manifest("/")
+        state.test_manifest = manifest.Manifest(state.sync["path"], "/")
```
Comment (jgraham) [inline tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/update/sync.py:103]: Is this different to `state.tests_path`?

Comment (jgraham) [reply tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/update/sync.py:103]: I guess we use it with `manifest.update` so it seem fine.

**Review by jgraham — APPROVED** [review]
Thanks for fixing this. If you wanted to make more changes, replacing manifest.update with manifest.load_and_update might be a good idea.

File: tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/update/sync.py (hunk 1 above)
Comment (jdm) [reply tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/update/sync.py:103]: When I used state.tests_path, I got exceptions for files that existed in the sync tree and not in the pre-sync tree.

**Review by jgraham — APPROVED** [review]

**Comment by jdm** [comment]
@bors-servo retry
* #24786

**Comment by jdm** [comment]
@bors-servo retry
* #23290

**Pull request merged**